Sherlock/test.cc

```diff
@@ -185,18 +220,22 @@ TEST(Sherlock, SubscribeAsynchronously) {
   using namespace sherlock_unittest;
 
   auto bar_stream = current::sherlock::Stream<Record>();
+  current::time::SetNow(std::chrono::microseconds(40));
   bar_stream.Publish(4);
+  current::time::SetNow(std::chrono::microseconds(50));
   bar_stream.Publish(5);
+  current::time::SetNow(std::chrono::microseconds(60));
   bar_stream.Publish(6);
   Data d;
-  std::unique_ptr<SherlockTestProcessor> p(new SherlockTestProcessor(d, false));
+  std::unique_ptr<SherlockTestProcessor> p(new SherlockTestProcessor(d, false, true));
   p->SetMax(4u);
   bar_stream.Subscribe(std::move(p)).Detach();  // `.Detach()` results in the subscriber running on its own.
   while (d.seen_ < 3u) {
     ;  // Spin lock.
   }
   EXPECT_EQ(3u, d.seen_);
-  EXPECT_EQ("4,5,6", d.results_);  // No `TERMINATE` for an asyncronous subscriber.
+  std::vector<std::string> expected_values{"[0:40,2:60] 4", "[1:50,2:60] 5", "[2:60,2:60] 6"};
```

Comment:
But you don't use `std::move()` anywhere for those vars, or what am I missing? :)